bool right
bool up
bool left
bool radius
